rollup.config.js

```diff
@@ -0,0 +1,70 @@
+  import vue from 'rollup-plugin-vue';
+  import swc from 'rollup-plugin-swc';
+  import replace from "@rollup/plugin-replace";
+  import commonjs from '@rollup/plugin-commonjs';
+  import { nodeResolve } from '@rollup/plugin-node-resolve';
+
+import pkg from './package.json';
+
+const extensions = ['.js', '.ts'];
+const external = _ => /node_modules/.test(_) && !/@swc\/helpers/.test(_);
+
+const plugins = targets => [
+  nodeResolve({
+    extensions
+  }),
+  replace({
+    preventAssignment: true,
+    values: {
+      'import.meta.env.VERSION': JSON.stringify(pkg.version)
+    }
+  }),
+  swc({
+    env: {
+      targets
+    },
+    module: {
+      type: "es6"
+    },
+    sourceMaps: true
+  })
+]
+
+export default [
+  {
+    external,
+    input: 'src/index.js',
+    output: {
+      format: 'esm',
+      file: pkg.module
+    },
+    plugins: [
+      vue(),
+      ...plugins("defaults and supports es6-module"),
+    ]
+  },
+  {
+    external,
+    input: 'src/index.js',
+    output: {
+      format: 'cjs',
+      file: pkg.main
+    },
+    plugins: [
+      vue({ template: { optimizeSSR: true } }),
```

Comment:
lets move it to `plugins()`. We can add second optional parameter to it to pass this options.